device/cluster.cpp

```diff
@@ -1000,6 +1012,15 @@ void Cluster::deassert_risc_reset_at_core(tt_cxy_pair core, const TensixSoftRese
     }
 }
 
+void Cluster::deassert_risc_reset_at_core(
+    const chip_id_t chip, const CoreCoord core, const TensixSoftResetOptions& soft_resets) {
+    tt_cxy_pair virtual_core;
+    const CoreCoord virtual_coord = to(chip, core, CoordSystem::VIRTUAL);
+    virtual_core.x = virtual_coord.x;
+    virtual_core.y = virtual_coord.y;
```

Comment:
Consider adding "get_xy_pair" to CoreCoord class, which would eliminate the need to do this. And then change everywhere in this class.

Reply:
This is tt_cxy_pair, so it needs chip as well which we don't have in CoreCoord. Idea is to use (chip, CoreCoord) instead of tt_cxy_pair

Reply:
Ah, it's the opposite case.
In that case you can do deassert_risc_reset_at_core({chip , to(chip, core, CoordSystem::VIRTUAL)}, soft_resets);

Reply:
to is deleted from Cluster, there was a comment to have translation only on Soc desc level. And it would return CoreCoord, but the old API expects tt_cxy_pair. I would keep this, we are going to dismantle this by removing old API